<launch>
  <arg name="tf_dynamic" default="true"/>
  <arg name="camera_link" default="head_rgbd_sensor_rgb_frame"/>
  <arg name="base_link" default="base_link"/>
  <arg name="tf_msg" default="/tf_msg/xtion_frame"/>
  <arg name="tf_base" default="/tf_msg/base_frame"/>

  <node name="dynamic_base_tf" pkg="roboenvcv" type="dynamic_sensor_tfstamped">
        <param name="dynamic_frame" value="$(arg base_link)"/>
        <remap from="/tf_msg/dynamic_sensor" to="$(arg tf_base)"/>
  </node>

  <group if="$(arg tf_dynamic)">
    <node name="dynamic_sensor_tf" pkg="roboenvcv" type="dynamic_sensor_tfstamped">
          <param name="dynamic_frame" value="$(arg camera_link)"/>
          <remap from="/tf_msg/dynamic_sensor" to="$(arg tf_msg)"/>
    </node>
  </group>
</launch>
